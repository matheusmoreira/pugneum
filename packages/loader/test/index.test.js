'use strict';

const fs = require('fs');
const path = require('path');
const assert = require('assert');
const walk = require('pugneum-walk');
const lex = require('pugneum-lexer');
const parse = require('pugneum-parser');
const load = require('../');

test('pugneum-loader', () => {
  let filename = __dirname + '/foo.pg';
  let source = fs.readFileSync(filename, 'utf8');
console.log(source);
  let tokens = lex(source, {filename});
  let ast = parse(tokens, {filename});

  ast = load(ast, {lex, parse});

  ast = walk(
    ast,
    function(node) {
      if (node.filename)
        node.filename = path.basename(node.filename);
      if (node.fullPath)
        node.fullPath = path.basename(node.fullPath);
    },
    {includeDependencies: true}
  );

  expect(ast).toMatchSnapshot();
});
