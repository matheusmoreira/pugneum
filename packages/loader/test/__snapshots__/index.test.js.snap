// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pugneum-loader 1`] = `
{
  "filename": "foo.pg",
  "line": 0,
  "nodes": [
    {
      "column": 1,
      "file": {
        "ast": {
          "filename": "bar.pg",
          "line": 0,
          "nodes": [
            {
              "column": 1,
              "filename": "bar.pg",
              "line": 1,
              "mode": "replace",
              "name": "bing",
              "nodes": [],
              "type": "NamedBlock",
            },
          ],
          "type": "Block",
        },
        "column": 9,
        "filename": "foo.pg",
        "fullPath": "bar.pg",
        "line": 1,
        "path": "bar.pg",
        "raw": "block bing
",
        "str": "block bing
",
        "type": "FileReference",
      },
      "filename": "foo.pg",
      "line": 1,
      "type": "Extends",
    },
    {
      "column": 1,
      "filename": "foo.pg",
      "line": 3,
      "mode": "replace",
      "name": "bing",
      "nodes": [
        {
          "block": {
            "filename": "foo.pg",
            "line": 4,
            "nodes": [],
            "type": "Block",
          },
          "column": 3,
          "file": {
            "ast": {
              "filename": "bing.pg",
              "line": 0,
              "nodes": [
                {
                  "attributeBlocks": [],
                  "attrs": [
                    {
                      "column": 1,
                      "filename": "bing.pg",
                      "line": 1,
                      "mustEscape": false,
                      "name": "class",
                      "val": "'bing'",
                    },
                  ],
                  "block": {
                    "filename": "bing.pg",
                    "line": 1,
                    "nodes": [
                      {
                        "column": 7,
                        "filename": "bing.pg",
                        "line": 1,
                        "type": "Text",
                        "val": "bong",
                      },
                    ],
                    "type": "Block",
                  },
                  "column": 1,
                  "filename": "bing.pg",
                  "isInline": false,
                  "line": 1,
                  "name": "div",
                  "selfClosing": false,
                  "type": "Tag",
                },
              ],
              "type": "Block",
            },
            "column": 11,
            "filename": "foo.pg",
            "fullPath": "bing.pg",
            "line": 4,
            "path": "bing.pg",
            "raw": ".bing bong
",
            "str": ".bing bong
",
            "type": "FileReference",
          },
          "filename": "foo.pg",
          "line": 4,
          "type": "Include",
        },
        {
          "attributeBlocks": [],
          "attrs": [],
          "block": {
            "filename": "foo.pg",
            "line": 5,
            "nodes": [
              {
                "column": 5,
                "file": {
                  "column": 13,
                  "filename": "foo.pg",
                  "fullPath": "script.js",
                  "line": 6,
                  "path": "script.js",
                  "raw": "document.write('hello world!');
",
                  "str": "document.write('hello world!');
",
                  "type": "FileReference",
                },
                "filename": "foo.pg",
                "filters": [],
                "line": 6,
                "type": "RawInclude",
              },
            ],
            "type": "Block",
          },
          "column": 3,
          "filename": "foo.pg",
          "isInline": false,
          "line": 5,
          "name": "script",
          "selfClosing": false,
          "type": "Tag",
        },
      ],
      "type": "NamedBlock",
    },
  ],
  "type": "Block",
}
`;
